### ISSUE TITLE
Conda crashes when updating cryptography

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda#2538]
When I type `conda update cryptography`, I get:
```
Traceback (most recent call last):
  File "/home/ilan/a/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/home/ilan/conda/conda/cli/main.py", line 120, in main
    args_func(args, p)
  File "/home/ilan/conda/conda/cli/main.py", line 127, in args_func
    args.func(args, p)
  File "/home/ilan/conda/conda/cli/main_install.py", line 62, in execute
    install.install(args, parser, 'install')
  File "/home/ilan/conda/conda/cli/install.py", line 394, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "/home/ilan/conda/conda/plan.py", line 556, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/home/ilan/conda/conda/instructions.py", line 133, in execute_instructions
    cmd(state, arg)
  File "/home/ilan/conda/conda/instructions.py", line 80, in UNLINK_CMD
    install.unlink(state['prefix'], arg)
  File "/home/ilan/conda/conda/install.py", line 1059, in unlink
    mk_menus(prefix, meta['files'], remove=True)
TypeError: 'NoneType' object has no attribute '__getitem__'
```
The output from `conda info` is:
```
Using Anaconda Cloud api site https://api.anaconda.org
Current conda install:

             platform : linux-64
        conda version : 4.0.6.dev404+e98fce7
  conda-build version : 1.20.3
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /home/ilan/a  (writable)
  default environment : /home/ilan/a
     envs directories : /home/ilan/a/envs
        package cache : /home/ilan/a/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/ilan/.condarc
    is foreign system : False
```

**Comment by ilanschnell** [comment]
CC @kalefranz

**Comment by mcg1969** [comment]
@kalefranz, @msarahan, @mingwandroid : do any of you know if any of the other `conda` projects rely on `load_meta`? (e.g., conda-build, conda-env, etc.) Within `conda`, this is the only place that uses it; and if I can assume that, the fix will be straightforward.

**Comment by mcg1969** [comment]
@ilanschnell, I'm actually unable to reproduce this simply by typing `conda update cryptography` in my root environment. By chance did you install it from a different channel?

**Comment by ilanschnell** [comment]
Yes, this is probably the cause of this bug, the fact that the package was first installed from a different channel.

**Comment by mcg1969** [comment]
I tried installing a fresh Miniconda 2 install on Linux. Then I did
``
conda install -c Massimo cryptography
```
which gave me a version of `cryptography 1.1.1`. Then I added `Massimo` to the *end* of my channel list, so it has lower priority, and I did
```
conda --debug cryptography
```
and here's the key output:
```
DEBUG:conda.instructions: PREFIX('/home/mcgrant/miniconda2')
DEBUG:conda.instructions: PRINT('Unlinking packages ...')   
Unlinking packages ...                                      
INFO:print:Unlinking packages ...                           
DEBUG:conda.instructions: PROGRESS('1')                     
INFO:progress.start:1                                       
DEBUG:conda.instructions: UNLINK(u'Massimo::cryptography-1.1.1-py27_0')
INFO:progress.update:(u'cryptography', 0)                              
[      COMPLETE      ]|#############################################################################| 100%
INFO:progress.stop:None                                                                                   
DEBUG:conda.instructions: PRINT('Linking packages ...')                                                   
Linking packages ...                                                                                      
INFO:print:Linking packages ...                                                                           
DEBUG:conda.instructions: PROGRESS('1')                                                                   
INFO:progress.start:1                                                                                     
DEBUG:conda.instructions: LINK(u'cryptography-1.3.2-py27_0 1')                                            
INFO:progress.update:(u'cryptography', 0)                                                                 
DEBUG:conda.install:pkgs_dir='/home/mcgrant/miniconda2/pkgs', prefix='/home/mcgrant/miniconda2', dist=u'cryptography-1.3.2-py27_0', linktype=1                                                                      
[      COMPLETE      ]|#############################################################################| 100%
```
So I'm having a bit of trouble reproducing the issue. If you can identify a reliable way to do so, please let me know. I'll keep digging

**Comment by ilanschnell** [comment]
So, I'm still seeing this issue on one of my environments.

**Comment by mcg1969** [comment]
Aha! This is the same problems as #2599. Fixed in #2638

**Comment by kalefranz** [comment]
Cool!

**Comment by ilanschnell** [comment]
I can confirm that it has been fixed (on `4.0.8-520-g874e8a7`).  Good job!
